app/models/equipment_object.rb

```diff
@@ -73,39 +73,90 @@ def make_switch_notes(old_res, new_res, handler)
-          new_val = diff[1] ? EquipmentModel.find(diff[1]).name : 'nil'
+      # write notes header
+      header = "### Edited on #{Time.current.to_s(:long)} by "\
+        "#{current_user.md_link}\n"
+      self.notes = notes ? notes + "\n" + header : header
+      # add notes if they exist
+      self.notes += "\n\n#### Notes:\n"\
+      "#{new_notes}" unless new_notes.empty?
+
+      # record changes
+      # rubocop:disable BlockNesting
+      changes = changes.delete_if { |x| x == 'deactivation_reason' }
+      unless changes.empty?
+        self.notes += "\n\n#### Changes:"
+        changes.each do |param, diff|
+          case param
+          when 'name'
+            name = 'Name'
+            old_val = diff[0].to_s
+            new_val = diff[1].to_s
+          when 'serial'
+            name = 'Serial'
+            old_val = diff[0].to_s
+            new_val = diff[1].to_s
+          when 'equipment_model_id'
+            name = 'EquipmentModel'
+            old_val = diff[0] ? EquipmentModel.find(diff[0]).name : 'nil'
+            new_val = diff[1] ? EquipmentModel.find(diff[1]).name : 'nil'
+          end
+          self.notes += "\n#{name} changed from " + old_val + ' to '\
+            + new_val + '.'
         end
-        new_notes += "\n#{name} changed from " + old_val + ' to ' + new_val\
-          + '.' if old_val && new_val
       end
+      # rubocop:enable BlockNesting
+      self.notes = self.notes.strip
+      self
     end
-    new_notes += "\n\n" + notes
-    self.notes = new_notes.strip
-    self
   end
 
+  # This is the old code for updating. I changed it to the above code
+  # to be more consistent with the logic used in the reservation
+  # controller's update action. This old code seemed incomplete...
+  # Notes were not correctly updated, and in fact the update action was
+  # missing the "new notes" parameter entirely.
+
+  # def update(current_user, new_params)
+  #   assign_attributes(new_params)
+  #   changes = self.changes
+  #   if changes.empty?
+  #     return self
+  #   else
+  #     new_notes = "#### Edited at #{Time.current.to_s(:long)} by "\
+  #       "#{current_user.md_link}\n\n"
+  #     new_notes += "\n\n#### Changes:"
+  #     changes.each do |param, diff|
+  #       case param
+  #       when 'name'
+  #         name = 'Name'
+  #         old_val = diff[0].to_s
+  #         new_val = diff[1].to_s
+  #       when 'serial'
+  #         name = 'Serial'
+  #         old_val = diff[0].to_s
+  #         new_val = diff[1].to_s
+  #       when 'equipment_model_id'
+  #         name = 'Equipment Model'
+  #         old_val = diff[0] ? EquipmentModel.find(diff[0]).name : 'nil'
+  #         new_val = diff[1] ? EquipmentModel.find(diff[1]).name : 'nil'
+  #       end
+  #       new_notes += "\n#{name} changed from " + old_val + ' to ' + new_val\
+  #         + '.' if old_val && new_val
+  #     end
+  #   end
+  #   new_notes += "\n\n" + notes
```

Comment:
This is where we were prepending the notes for Equipment Items (instead of appending them); when you copied over the `update` method from the Reservation model you switched over to appending notes.